src/test/java/redis/clients/jedis/tests/commands/PublishSubscribeCommandsTest.java

```diff
@@ -301,6 +338,74 @@ public void onPMessage(byte[] pattern, byte[] channel, byte[] message) {
     }, SafeEncoder.encode("foo.*"), SafeEncoder.encode("bar.*"));
   }
 
+  @Test
+  public void binaryPubSubChannelWithPingPong() {
+    final CountDownLatch latchUnsubscribed = new CountDownLatch(1);
+    final CountDownLatch latchReceivedPong = new CountDownLatch(1);
+
+    jedis.subscribe(new BinaryJedisPubSub() {
+
+      @Override
+      public void onSubscribe(byte[] channel, int subscribedChannels) {
+        publishOne("testchan1", "hello");
+      }
+
+      @Override
+      public void onMessage(byte[] channel, byte[] message) {
+        this.ping();
+      }
+
+      @Override
+      public void onPong(byte[] pattern) {
+        latchReceivedPong.countDown();
+        unsubscribe();
+      }
+
+      @Override
+      public void onUnsubscribe(byte[] channel, int subscribedChannels) {
+        latchUnsubscribed.countDown();
+      }
+    }, SafeEncoder.encode("testchan1"));
+    assertEquals(0L, latchReceivedPong.getCount());
+    assertEquals(0L, latchUnsubscribed.getCount());
+  }
+
+  @Test
+  public void binaryPubSubChannelWithPingPongWithArgument() {
+    final CountDownLatch latchUnsubscribed = new CountDownLatch(1);
+    final CountDownLatch latchReceivedPong = new CountDownLatch(1);
+    final List<byte[]> pongPatterns = new ArrayList<>();
+
+    jedis.subscribe(new BinaryJedisPubSub() {
+
+      @Override
+      public void onSubscribe(byte[] channel, int subscribedChannels) {
+        publishOne("testchan1", "hello");
+      }
+
+      @Override
+      public void onMessage(byte[] channel, byte[] message) {
+        this.ping("hi!".getBytes());
```

Comment:
Instread of String.getBytes(), use SafeEncoder.encode(String).

Alternate idea, keep SafeEncoder.encode(message) in variable and use it wherever you want.

Reply:
That's a good idea!  Just updated it.